src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js

```diff
@@ -13,15 +13,27 @@ const generateGeolocationString = (location = {}) => {
     loc('geolocation.formatting.partial', 'login', [location.city, location.country]);
 };
 
+// A map from FactorTransactionIntent in okta-core to customer-facing flow name
+const challengeIntentToFlowMap = {
+  'AUTHENTICATION': loc('idx.return.link.otponly.enter.code.on.page.sign.in', 'login'),
+  'RECOVERY': loc('idx.return.link.otponly.enter.code.on.page.password.reset', 'login'),
+  'UNLOCK_ACCOUNT': loc('idx.return.link.otponly.enter.code.on.page.account.unlock', 'login'),
+  'ENROLLMENT': loc('idx.return.link.otponly.enter.code.on.page.registration', 'login')
+};
+
 const getTerminalOtpEmailMagicLinkContext = (settings, appState) => {
   const app = appState.get('app');
   const client = appState.get('client');
-  let appName, browserOnOsString, geolocation;
+  const challengeIntent = challengeIntentToFlowMap[appState.get('idx').context.intent];
+  let enterCodeOnFlowPage, appName, browserOnOsString, isMobileDevice, geolocation;
+  enterCodeOnFlowPage = loc('idx.return.link.otponly.enter.code.on.page', 'login', [challengeIntent]);
   if (app) {
-    appName = loc('idx.return.link.otponly.accessing.app', 'login', [app.label]);
+    appName = loc('idx.return.link.otponly.app', 'login', [app.label]);
   }
   if (client) {
     browserOnOsString = loc('idx.return.link.otponly.browser.on.os', 'login', [client.browser, client.os]);
+    // TODO: check exact name
```

Comment:
Just a check, do you have a ticket created for this?

Reply:
Good catch, designer and PM are ok with my approach (considering any device whose OS name contains "Android" or "iOS" as mobile device), I will delete this TODO.